assets/js/event-providers/woocommerce.js

```diff
@@ -19,13 +19,171 @@
 		return;
 	}
 
+	const currency = global._googlesitekit.wcdata.currency;
+
+	if ( global._googlesitekit.wcdata.add_to_cart ) {
+		const { price } = global._googlesitekit.wcdata.add_to_cart;
+
+		const eventData = formatEventData(
+			price,
+			currency,
+			global._googlesitekit.wcdata.add_to_cart
+		);
+
+		global._googlesitekit?.gtagEvent?.( 'add_to_cart', eventData );
+	}
+
+	if ( global._googlesitekit.wcdata.purchase ) {
+		const { id, totals, items } = global._googlesitekit.wcdata.purchase;
+
+		const eventData = formatEventData(
+			totals.total_price,
+			totals.currency_code,
+			items,
+			id,
+			totals.shipping_total,
+			totals.tax_total
+		);
+
+		global._googlesitekit?.gtagEvent?.( 'purchase', eventData );
+	}
+
 	const body = jQuery( 'body' );
 
-	body.on( 'added_to_cart', () => {
-		global._googlesitekit?.gtagEvent?.( 'add_to_cart' );
-	} );
+	// eslint-disable-next-line camelcase
+	body.on( 'added_to_cart', ( event, fragments, cart_hash, $button ) => {
+		const productID = parseInt( $button.data( 'product_id' ), 10 );
+		const productData =
+			global._googlesitekit.wcdata.products?.find(
+				( product ) => product?.id === productID
+			) || {};
+		const { price } = productData;
 
-	body.on( 'checkout_place_order_success', () => {
-		global._googlesitekit?.gtagEvent?.( 'purchase' );
+		const eventData = formatEventData( price, currency, productData );
+		global._googlesitekit?.gtagEvent?.( 'add_to_cart', eventData );
 	} );
+
+	document
+		.querySelectorAll(
+			'.products-block-post-template .product, .wc-block-product-template .product'
+		)
+		?.forEach( ( productCard ) => {
+			// Get the Product ID from a child node containing the relevant attribute.
+			const productID = parseInt(
+				productCard
+					.querySelector( '[data-product_id]' )
+					?.getAttribute( 'data-product_id' ),
+				10
+			);
+
+			if ( ! productID ) {
+				return;
+			}
+
+			productCard.addEventListener( 'click', ( event ) => {
+				const target = event.target;
+				const button = target.closest(
+					'.wc-block-components-product-button [data-product_id]'
+				);
+
+				const isAddToCartButton =
+					button &&
+					button.classList.contains( 'add_to_cart_button' ) &&
+					! button.classList.contains( 'product_type_variable' );
+
+				if ( isAddToCartButton ) {
+					const productData =
+						global._googlesitekit.wcdata.products?.find(
```

Comment:
Let's destruct `global._googlesitekit.wcdata` at the beginning to avoid repeating it everywhere in this file.